stdlib/src/hashlib.rs

```diff
@@ -211,108 +213,144 @@ mod hashlib {
+    pub fn local_sha384(args: HashArgs) -> PyHasher {
         PyHasher::new("sha384", HashWrapper::new::<Sha384>(args.string))
     }
 
-    #[pyfunction]
-    fn sha512(args: HashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_sha512")]
+    pub fn local_sha512(args: HashArgs) -> PyHasher {
         PyHasher::new("sha512", HashWrapper::new::<Sha512>(args.string))
     }
 
-    #[pyfunction]
-    fn sha3_224(args: HashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_sha3_224")]
+    pub fn local_sha3_224(args: HashArgs) -> PyHasher {
         PyHasher::new("sha3_224", HashWrapper::new::<Sha3_224>(args.string))
     }
 
-    #[pyfunction]
-    fn sha3_256(args: HashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_sha3_256")]
+    pub fn local_sha3_256(args: HashArgs) -> PyHasher {
         PyHasher::new("sha3_256", HashWrapper::new::<Sha3_256>(args.string))
     }
 
-    #[pyfunction]
-    fn sha3_384(args: HashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_sha3_384")]
+    pub fn local_sha3_384(args: HashArgs) -> PyHasher {
         PyHasher::new("sha3_384", HashWrapper::new::<Sha3_384>(args.string))
     }
 
-    #[pyfunction]
-    fn sha3_512(args: HashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_sha3_512")]
+    pub fn local_sha3_512(args: HashArgs) -> PyHasher {
         PyHasher::new("sha3_512", HashWrapper::new::<Sha3_512>(args.string))
     }
 
-    #[pyfunction]
-    fn shake_128(args: HashArgs) -> PyHasherXof {
+    #[pyfunction(name = "openssl_shake_128")]
+    pub fn local_shake_128(args: HashArgs) -> PyHasherXof {
         PyHasherXof::new("shake_128", HashXofWrapper::new_shake_128(args.string))
     }
 
-    #[pyfunction]
-    fn shake_256(args: HashArgs) -> PyHasherXof {
+    #[pyfunction(name = "openssl_shake_256")]
+    pub fn local_shake_256(args: HashArgs) -> PyHasherXof {
         PyHasherXof::new("shake_256", HashXofWrapper::new_shake_256(args.string))
     }
 
-    #[pyfunction]
-    fn blake2b(args: BlakeHashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_blake2b")]
+    pub fn local_blake2b(args: BlakeHashArgs) -> PyHasher {
         PyHasher::new("blake2b", HashWrapper::new::<Blake2b512>(args.data))
     }
 
-    #[pyfunction]
-    fn blake2s(args: BlakeHashArgs) -> PyHasher {
+    #[pyfunction(name = "openssl_blake2s")]
+    pub fn local_blake2s(args: BlakeHashArgs) -> PyHasher {
         PyHasher::new("blake2s", HashWrapper::new::<Blake2s256>(args.data))
     }
 
-    trait ThreadSafeDynDigest: DynClone + DynDigest + Sync + Send {}
+    #[pyfunction]
+    fn compare_digest(
+        a: ArgStrOrBytesLike,
+        b: ArgStrOrBytesLike,
+        vm: &VirtualMachine,
+    ) -> PyResult<PyObjectRef> {
+        if a.get_enum_int() != b.get_enum_int() {
+            return Err(vm.new_type_error(format!(
+                "different types in compare_digest {} and {} ",
+                a.get_enum(),
+                b.get_enum()
+            )));
+        }
```

Comment:
The error message is different.
I am not going to ask to match the message matching, but I disagree about adding `get_enum` and `get_enum_int` to make incompatible error message.
```
Python 3.11.1 (main, Feb  4 2023, 11:11:18) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
>>> _hashlib.compare_digest('', bytearray(b''))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
```

```suggestion
        fn is_str(arg: &ArgStrOrBytesLike) -> {
           matches!(arg, ArgStrOrBytesLike::String(_))
        }
        if is_str(&a) != is_str(&b) {
            return Err(...);
        }
```

Reply:
@youknowone are you suggesting comparing class names as strings character by character? It takes a lot of time on performance.

Reply:
I agree that method that i used is disgusting, but i didn't found any examples by comparing instances enums.

Reply:
I left a code suggestion using matches! macro